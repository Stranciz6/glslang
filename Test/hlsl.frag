float4x4 World;
float4x4 View;
float4x4 Projection;

float4 AmbientColor = float4(1, 0.5, 0, 1);
float AmbientIntensity = 0.1;

//float4 PixelShaderFunction(VertexShaderOutput input) : COLOR0
//{
//    return AmbientColor * AmbientIntensity;
//}
