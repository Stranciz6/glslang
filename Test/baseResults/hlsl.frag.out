hlsl.frag
Shader version: 100
gl_FragCoord origin is upper left
0:? Sequence
0:5  move second child to first child (temp 4-component vector of float)
0:5    'AmbientColor' (temp 4-component vector of float)
0:?     Constant:
0:?       1.000000
0:?       0.500000
0:?       0.000000
0:?       1.000000
0:6  move second child to first child (temp float)
0:6    'AmbientIntensity' (temp float)
0:6    Constant:
0:6      0.100000
0:?   Linker Objects
0:?     'World' (temp 4X4 matrix of float)
0:?     'View' (temp 4X4 matrix of float)
0:?     'Projection' (temp 4X4 matrix of float)
0:?     'AmbientColor' (temp 4-component vector of float)
0:?     'AmbientIntensity' (temp float)


Linked fragment stage:


Shader version: 100
gl_FragCoord origin is upper left
0:? Sequence
0:5  move second child to first child (temp 4-component vector of float)
0:5    'AmbientColor' (temp 4-component vector of float)
0:?     Constant:
0:?       1.000000
0:?       0.500000
0:?       0.000000
0:?       1.000000
0:6  move second child to first child (temp float)
0:6    'AmbientIntensity' (temp float)
0:6    Constant:
0:6      0.100000
0:?   Linker Objects
0:?     'World' (temp 4X4 matrix of float)
0:?     'View' (temp 4X4 matrix of float)
0:?     'Projection' (temp 4X4 matrix of float)
0:?     'AmbientColor' (temp 4-component vector of float)
0:?     'AmbientIntensity' (temp float)

// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 17

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "PixelShaderFunction"
                              ExecutionMode 4 OriginUpperLeft
                              Source HLSL 100
                              Name 4  "PixelShaderFunction"
                              Name 10  "World"
                              Name 11  "View"
                              Name 12  "Projection"
                              Name 14  "AmbientColor"
                              Name 16  "AmbientIntensity"
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypeVector 6(float) 4
               8:             TypeMatrix 7(fvec4) 4
               9:             TypePointer Function 8
              13:             TypePointer Function 7(fvec4)
              15:             TypePointer Function 6(float)
4(PixelShaderFunction):           2 Function None 3
               5:             Label
       10(World):      9(ptr) Variable Function
        11(View):      9(ptr) Variable Function
  12(Projection):      9(ptr) Variable Function
14(AmbientColor):     13(ptr) Variable Function
16(AmbientIntensity):     15(ptr) Variable Function
                              FunctionEnd
